pkg/controllers/resources/ingressclasses/syncer.go

```diff
@@ -43,7 +49,7 @@ func (i *ingressClassSyncer) Resource() client.Object {
 var _ syncertypes.Syncer = &ingressClassSyncer{}
 
 func (i *ingressClassSyncer) Syncer() syncertypes.Sync[client.Object] {
-	return syncer.ToGenericSyncer(i)
+	return syncer.ToGenericSyncer[*networkingv1.IngressClass](i)
```

Comment:
Thats not needed

Reply:
That was my IDE complaining. I'll remove it then.